decidim-core/app/mailers/decidim/application_mailer.rb

```diff
@@ -5,8 +5,23 @@ module Decidim
   # mailers.
   class ApplicationMailer < ActionMailer::Base
     include LocalisedMailer
+    after_action :set_smtp
 
     default from: Decidim.config.mailer_sender
     layout "decidim/mailer"
+
+    private
+
+    def set_smtp
+      return if @organization.nil? || @organization.smtp_settings.blank?
```

Comment:
There's a problem with just checking the settings are blank, I think we should validate them. If you update an organization at the system panel keep the SMTP form blank the organization has this as `smtp_settings`:

```ruby
{"from"=>"", "port"=>"", "address"=>"", "user_name"=>"", "encrypted_password"=>nil}
```

Which it isn't blank and would change the delivery method to something invalid.